### ISSUE TITLE
Parameter select does not show labels and properties for big database

### ISSUE THREAD

**Issue by dschmitz89** [nielsdejong/neodash#50]
Hello,

thank you for this awesome dashboard functionality. 

When I run a dedicated cypher query, I get nice chart (see image on left). However Parameter select does not show any options. I assume you are fetching them from a function like 'call apoc.meta.data()'. Unfortunately, that call also does not finish within a minute when I run it in neo4j browser. A simple call to find all node labels ("MATCH (n) RETURN distinct labels(n)") already takes half a minute for my case. Our DB contains 55 Mio. nodes and 100 Mio. relationships. Am I correct that slow query time is the problem here? And if, is there a way around maybe?

Thanks in advance!

<img width="901" alt="Dashboard_Selection_Problem" src="https://user-images.githubusercontent.com/40656107/147074928-79458ad4-30da-473c-9c40-3aaa7a7c985d.png">

**Comment by nielsdejong** [comment]
Hi @dschmitz89 , thank you for reporting this issue!
Does the `Node label` box also not show any suggestions after you start typing in it? It is by default waiting for you to specify some characters as to not create slow queries. Do you know how many distinct node labels you have in the graph, and how many different properties per node label?

In the case of this report, I'm calling `CALL db.schema.nodeTypeProperties()` via Cypher, which should be the most optimized way of reading the schema from the graph. Could you check now long that query takes for you in the Neo4j browser?

**Comment by dschmitz89** [comment]
Hi @nielsdejong ,

thanks for the quick reply!
`CALL db.schema.nodeTypeProperties()` did not finish within two minutes, then aborted it.

Also when I type the complete node label into the box, I cannot proceed. Any way around this? I guess this will be a common issue for DBs with many properties/node types.

![image](https://user-images.githubusercontent.com/40656107/147245537-ee500aec-26c3-4770-94df-313aca7a7319.png)

**Comment by nielsdejong** [comment]
Hi @dschmitz89 ,

I've just pushed a solution for this in the 2.0.4 release.
Under 'show advanced settings' on the property select report, there's a new setting called **"Enable Manual Label/Property Name Specification"**. Toggling this switch will allow you to specify your node label and property name manually (without the autocomplete functionality). Keep in mind this is case sensitive, and will require you to specify the exact name of the node and property yourself.

Let me know if this solves the issue for you,
Niels

**Comment by dschmitz89** [comment]
Hi @nielsdejong ,

thank you so much, works like a charm now :)!

Would it maybe be possible to implement the same functionality for relationship attributes in future?

**Comment by nielsdejong** [comment]
Great! Have added the relationship property selector to the feature list. Thank you.

**Comment by emigre459** [comment]
@nielsdejong Along a similar line, is there a way to force manual control of the parameter value in the Selection tile (e.g. when using it regularly)? In my use case, I have a publication authorship and citation graph and I've built a workflow out of multiple Neodash tiles that lets a user find an author's name, returns a Table tile of possible name matches and their node IDs, then allows the user to type the node ID into another Parameter Select tile that will ultimately feed into a query in a Graph tile. I'm finding that I can't enter a node ID that is being returned by my Table tile, as the dropdown never populates with it. 

Related question: is there a limit being applied to the result set used for populating that dropdown, such that my `Author` node isn't getting queried to find the `id` property I specified before the dropdown results stop returning?
